index.js

```diff
@@ -73,77 +81,65 @@ request.cookie = function (str) {
   return cookies.parse(str)
 }
 
-request.defaults = function (options, requester) {
+function wrap (method, options, requester) {
```

Comment:
`wrap` is a little vague, it took me a while to understand what this function was doing. Is there a better name here that could provide a little more context? `wrapDefaults`? `wrapRequestMethods`?

Also, some documentation on expected arguments would be equally helpful, especially since it looks like we support a lot of different ways to call this (method = string, function, or request, request is optional, etc).